.github/workflows/pio.yml

```diff
@@ -0,0 +1,44 @@
+name: PlatformIO CI
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  pio-build:
+    runs-on: ubuntu-20.04
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Cache pip
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
+        restore-keys: |
+          ${{ runner.os }}-pip-
+
+    - name: Cache PlatformIO
+      uses: actions/cache@v2
+      with:
+        path: ~/.platformio
+        key: ${{ runner.os }}-${{ hashFiles('**/lockfiles') }}
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
+
+    - name: Install PlatformIO
+      run: |
+        python -m pip install --upgrade pip
+        pip install --upgrade platformio
+
+    - name: Build Firmware
+      run: |
+        pio run -d ./arduino/splitflap \
+          -e uno-shift-register \
+          -e uno-direct \
+          -e mega-direct \
+          -e esp32 \
+          -e chainlink
```

Comment:
Since it's my first time dealing with GH Actions, is it better to split these into individual tasks?

Reply:
I suspect it will be easier to tell which failed if they're separate tasks, but let's start with a single task as-is for now and see how it goes for a while.